File: src/com/cattailsw/nanidroid/dlgs/UserInputDlg.java
package com.cattailsw.nanidroid.dlgs;

import android.support.v4.app.DialogFragment;
import android.os.Bundle;

import com.cattailsw.nanidroid.R;
import android.widget.EditText;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.view.View;
import android.widget.Button;

public class UserInputDlg extends DialogFragment {

    public interface UserInputListener {
	void onFinishUserInput(String id, String userinput);
	void onCancelInput();
    }

    private EditText mEdit = null;
    private Button okBtn = null;
    private Button cancleBtn = null;
    private String id = null;
    public UserInputDlg(String iid){
	id = iid;
    }

    public View onCreateView(final LayoutInflater layoutInflater, ViewGroup container, final Bundle bundle) {
	View v = layoutInflater.inflate(R.layout.dlg_user_input, container);
	mEdit = (EditText) v.findViewById(R.id.user_in);
	okBtn = (Button) v.findViewById(R.id.ok_btn);
	okBtn.setOnClickListener(new View.OnClickListener(){
		public void onClick(View v) {
		    UserInputListener i = (UserInputListener) getActivity();
		    i.onFinishUserInput(id, mEdit.getText().toString());
		    UserInputDlg.this.dismiss();
		}
	    });
	cancleBtn = (Button) v.findViewById(R.id.cancel_btn);
	cancleBtn.setOnClickListener(new View.OnClickListener(){
		public void onClick(View v) {
		    UserInputListener i = (UserInputListener) getActivity();
		    i.onCancelInput();
		    UserInputDlg.this.dismiss();
		}
	    });
	getDialog().setTitle(R.string.user_input_dlg_title);
	return v;
    }

    

}

File: src/com/cattailsw/nanidroid/dlgs/UserInputDlg.kt
package com.cattailsw.nanidroid.dlgs
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import com.cattailsw.nanidroid.R
class UserInputDlg() : DialogFragment() {
 interface UserInputListener { fun onFinishUserInput(id:String,userinput:String); fun onCancelInput() }
 private var edit:EditText?=null; private var id:String?=null
 constructor(id:String?) : this(){this.id=id}
 override fun onSaveInstanceState(out:Bundle){out.putString("id",id);super.onSaveInstanceState(out)}
 override fun onCreateView(inflater:LayoutInflater,container:ViewGroup?,state:Bundle?):View { if(id==null)id=state!!.getString("id");return inflater.inflate(R.layout.dlg_user_input,container).also{v->edit=v.findViewById(R.id.user_in);v.findViewById<Button>(R.id.ok_btn).setOnClickListener{(activity as UserInputListener).onFinishUserInput(id!!,edit!!.text.toString());dismiss()};v.findViewById<Button>(R.id.cancel_btn).setOnClickListener{(activity as UserInputListener).onCancelInput();dismiss()};dialog!!.setTitle(R.string.user_input_dlg_title)} }
}
